### PR TITLE
New🌟Add ultralytics analytics dashboard (Beta) 💙

### PR DESCRIPTION
**Pull request by RizwanMunawar**
### Key Features

- Added support for line graphs
- Implemented class-wise bar plots
- Introduced counting display feature

### CLI Command for Dashboard
```bash
# Build a dashboard with an ultralytics demo video
yolo dashboard

# Build dashboard with custom configuration
yolo dashboard "path/to/template.yaml"
```

### Python Code for Dashboard
```python
from ultralytics.cfg import handle_dashboard
        
# Build a dashboard with an ultralytics demo video
handle_dashboard()

# Build dashboard with custom configuration
handle_dashboard("path/to/template.yaml")
```
@glenn-jocher 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Introducing Ultralytics Analytics Dashboard: A Real-time Visualization and Analysis Tool for Object Detection.

### 📊 Key Changes
- **New Documentation**: Added comprehensive documentation for the analytics dashboard, explaining its configuration, custom settings, and usage.
- **Dashboard Configuration**: Simplified dashboard setup with a customizable YAML configuration file, supporting modes like predict, count, and track, and accommodating custom videos and model types (e.g., yolov8s, yolov8m, yolov8l).
- **Dynamic Visuals**: Real-time display of detection, tracking, and analytics through various visual representations including line graphs, bar plots, and object counting metrics within a user-friendly interface.
- **Extensible Analytics Framework**: The system's modular design ensures easy updates and the addition of new features in the future.

### 🎯 Purpose & Impact
- **Enhanced Monitoring**: Enables users to closely monitor and understand object detection and tracking activities, providing valuable insights into performance and trends.
- **Increased Accessibility**: Offers a straightforward and interactive way for both technical and non-technical users to visualize complex data generated by computer vision models.
- **Custom Visualization**: Supports custom configuration, allowing users to tailor the dashboard according to specific project requirements, thus enhancing the overall analysis experience.

This update makes deep learning analytics more accessible and interpretable, potentially benefiting a wide range of applications from retail analytics to traffic monitoring. 🚀🔍

### CODE REVIEW

**Review by Burhan-Q — COMMENTED** [review]
I didn't mess around with anything custom, just tested as is on Windows and Linux. Ran well overall, a few quick comments added as suggestions.

**Comment by Burhan-Q** [comment]
One additional idea, not sure how feasible it is, but wondering if there could be a more graceful way to terminate the example? Like adding a "Stop" or "Shutdown" example that would gracefully halt the program and cleanly exit.

**Comment by RizwanMunawar** [comment]
Ah, I got it @Burhan-Q Thanks for the updates :)

Yes definitely your suggested changes make sense, I am working on the streamlit side, and once done I will adjust these changes in that :)

File: ultralytics/cfg/__init__.py [hunk 1]
```diff
@@ -414,6 +414,17 @@ def handle_explorer():
...
 
+def handle_dashboard(yaml_filepath=ROOT / "cfg/analytics/dashboard.yaml"):
+    """
+    Open the Ultralytics Dashboard.
+
+    Args:
+        yaml_filepath (str): dashboard configuration YAML path
+    """
+    LOGGER.info("Ultralytics dashboard ... 😍")
+    subprocess.run(["python", ROOT / "analytics/dashboard.py", yaml_filepath])
```
[the remark this answers is not in the document]
Comment (RizwanMunawar) [reply ultralytics/cfg/__init__.py:425]: Yes, make sense, I will update it :)

File: ultralytics/analytics/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Ultralytics YOLO 🚀, AGPL-3.0 license
+# Documentation: https://docs.ultralytics.com/analytics/index.md
+# Example usage: yolo dashboard "path/to/custom_data.yaml"
+
+import os
+import sys
+import time
+
+import cv2
+from flask import Flask, Response, jsonify, render_template
```
[the remark this answers is not in the document]
Comment (RizwanMunawar) [reply ultralytics/analytics/dashboard.py:10]: Ah got it, you mean as we did for solution modules?

**Pull request closed without merging**